core/templates/pages/blog-dashboard-page/modal-templates/upload-blog-post-thumbnail.component.ts

```diff
@@ -73,8 +74,29 @@ export class UploadBlogPostThumbnailComponent implements OnInit {
     this.invalidImageWarningIsShown = false;
     let reader = new FileReader();
     reader.onload = (e) => {
+      this.invalidTagsAndAttributes = {
+        tags: [],
+        attrs: []
+      };
+      let imageData = (e.target as FileReader).result as string;
+      const mimeType = imageData.split(';')[0];
+      if (mimeType === 'data:image/svg+xml') {
+        let svg = this.svgSanitizerService.parseDataURI(imageData);
+        this.invalidTagsAndAttributes = (
+          this.svgSanitizerService.getInvalidSvgTagsAndAttrs(svg));
+        const tags = this.invalidTagsAndAttributes.tags;
+        let attrs: string[] = [];
+        this.invalidTagsAndAttributes.attrs.forEach(attribute => {
```

Comment:
Too much repeated code. Try not to repeat code in multiple components, it makes maintenance difficult. Centralize common code in the relevant service, instead.